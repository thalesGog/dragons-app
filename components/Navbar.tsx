import styled from 'styled-components'

const Navbar = styled.nav`
  border-bottom: 1px solid #10162f;
  background-color: #fff;
  position: sticky;
  width: 100%;
  top: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding-bottom: 4px;
  ul {
    margin: 0;
    padding: 0;
    list-style-type: none;
    display: flex;
    li {
      padding: 8px;
      cursor: pointer;
      text-transform: uppercase;
      font-weight: 600;
      &:hover {
        color: ${({theme}) => theme.colors.secondary};
      }
    }
  }
  span {
    cursor: pointer;
    &:hover {
      color: ${({theme}) => theme.colors.secondary};
    }
  }
`

export default Navbar
