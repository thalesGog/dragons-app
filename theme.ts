import {DefaultTheme} from 'styled-components'

const theme: DefaultTheme = {
  fontFamily: 'Roboto, sans-serif',
  backgroundColor: '#fff0e5',
  colors: {
    primary: '#10162f',
    secondary: '#3a10e5',
  },
}

export default theme
