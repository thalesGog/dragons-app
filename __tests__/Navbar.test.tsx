import React from 'react'
import {ThemeProvider} from 'styled-components'
import theme from '../theme'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import Navbar from '../components/Navbar'

test('Render Navbar component', async () => {
  const {container} = render(
    <ThemeProvider theme={theme}>
      <Navbar />
    </ThemeProvider>,
  )
  expect(container).toMatchSnapshot()
})
