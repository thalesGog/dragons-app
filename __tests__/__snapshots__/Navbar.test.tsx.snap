// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navbar component 1`] = `
.c0 {
  border-bottom: 1px solid #10162f;
  background-color: #fff;
  position: -webkit-sticky;
  position: sticky;
  width: 100%;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 4px;
}

.c0 ul {
  margin: 0;
  padding: 0;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 ul li {
  padding: 8px;
  cursor: pointer;
  text-transform: uppercase;
  font-weight: 600;
}

.c0 ul li:hover {
  color: #3a10e5;
}

.c0 span {
  cursor: pointer;
}

.c0 span:hover {
  color: #3a10e5;
}

.c0 img {
  padding-top: 8px;
}

<div>
  <nav
    class="c0"
  />
</div>
`;
